src/lib/utils/date.ts

```diff
@@ -1,2 +1,44 @@
 export const dateToString = (published: Date | string) =>
   new Date(published).toISOString().split("T")[0]
+
+export const formatDateRange = (startDate: string, endDate: string) => {
```

Comment:
Appreciate the work here... few things on this function.

I think we can simply use the `Intl.DateTimeFormat().formatRange` helper built into JavaScript, with the right options. It produces very similar strings to this function, but avoids the custom logic, and will better align with internationalization date-display standards, and offload any need to maintain this.

For example: `16-18 Aug` but also `Aug 23-Sep 13` (custom `formatDateRange`)
would become: `Aug 15 – 17` and `Aug 22 – Sep 12` (native `Intl.DateTimeFormat().formatRange`, with 2-digit day, numeric month, no year or time)

cc: @konopkja I think this may be _slightly_ different than your expressed preference, but I would ask to reconsider to avoid getting into customized ways to handle dates, when dealing with a very internationalized site. 

---

Also, in this custom `formatDateRange` we don't appear to be using `locale` (I see `undefined`) so it's defaulting to english.

In the end we should use `startDate`, `endDate` and `locale` with:

```ts
new Intl.DateTimeFormat(locale, { day: "2-digit", month: "short" }).formatRange(new Date(startDate), new Date(endDate))
```

...to get the result we're going for here.

```js
// Paste me into a node terminal / browser console to try it
var startDate = "2024-08-23"
var endDate = "2024-09-13"
var locale = "es-ES"
var shortDayMonth = new Intl.DateTimeFormat(locale, { day: "2-digit", month: "short" }).formatRange(new Date(startDate), new Date(endDate))
console.log(shortDayMonth) // 22 ago – 12 sept
```

Reply:
This makes complete sense, making the required modifications in date formats